src/slurm_plugin/computemgtd.py

```diff
@@ -0,0 +1,138 @@
+        )
+        self.clustermgtd_heartbeat_file_path = config.get("computemgtd", "clustermgtd_heartbeat_file_path")
+
+        proxy = config.get("computemgtd", "proxy", fallback=self.DEFAULTS.get("proxy"))
+        if proxy != "NONE":
+            self._boto3_config["proxies"] = {"https": proxy}
+        self.boto3_config = Config(**self._boto3_config)
+        self.logging_config = config.get("computemgtd", "logging_config", fallback=self.DEFAULTS.get("logging_config"))
+        # Log configuration
+        log.info(self.__repr__())
+
+
+def _self_terminate(computemgtd_config):
+    """Self terminate the instance."""
+    instance_manager = InstanceManager(
+        computemgtd_config.region, computemgtd_config.cluster_name, computemgtd_config.boto3_config
+    )
+    self_instance_id = check_command_output("curl http://169.254.169.254/latest/meta-data/instance-id")
+    log.info("Self terminating instance %s now!", self_instance_id)
+    instance_manager.delete_instances([self_instance_id])
+
+
+def _get_clustermgtd_heartbeat(clustermgtd_heartbeat_file_path):
+    """Get clustermgtd's last heartbeat."""
+    with open(clustermgtd_heartbeat_file_path, "r") as timestamp_file:
+        # Note: heartbeat must be written with datetime.strftime to convert localized datetime into str
+        # datetime.strptime will not work with str(datetime)
+        # Example timestamp written to heartbeat file: 2020-07-30 19:34:02.613338+00:00
+        return datetime.strptime(timestamp_file.read(), TIMESTAMP_FORMAT)
+
+
+def _run_computemgtd():
+    """Run computemgtd actions."""
+    # Initial default heartbeat time as computemgtd startup time
+    last_heartbeat = datetime.now(tz=timezone.utc)
+    log.info("Initializing clustermgtd heartbeat to be computemgtd startup time: %s", last_heartbeat)
+    while True:
+        # Get current time
+        current_time = datetime.now(tz=timezone.utc)
+        # Get program config
+        computemgtd_config = ComputemgtdConfig(os.path.join(CONFIG_FILE_DIR, "parallelcluster_computemgtd.conf"))
+        # Configure root logger
+        try:
+            fileConfig(computemgtd_config.logging_config, disable_existing_loggers=False)
+        except Exception as e:
+            log.warning(
+                "Unable to configure logging from %s, using default logging settings.\nException: %s",
+                computemgtd_config.logging_config,
+                e,
+            )
+        # Check heartbeat
+        try:
+            last_heartbeat = _get_clustermgtd_heartbeat(computemgtd_config.clustermgtd_heartbeat_file_path)
+            log.info("Latest heartbeat from clustermgtd: %s", last_heartbeat)
+        except Exception as e:
+            log.error("Unable to retrieve clustermgtd heartbeat with exception: %s", e)
+        finally:
+            # Any non-positive clustermgtd_timeout will disable self-termination
+            if computemgtd_config.clustermgtd_timeout > 0 and time_is_up(
```

Comment:
why don't we use a separate parameter to disable computemgtd functionalities and we check for it at the beginning of the loop so that we don't perform any other operation?